### PR TITLE
Auto correlation

### PR DESCRIPTION
**Pull request by idanstark42**

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -39,29 +40,29 @@ Logger.prototype.formatLog = function (level, moduleName, timestamp, correlation
     'level': LOG_LEVELS.LOG_LEVELS_TRANSLATION[level],
     'module': moduleName,
     'timestamp': timestamp.toISOString(),
-    'correlationId': correlationId,
+    'correlationId': getCorrelationId(),
```
Comment (taltaub22) [inline lib/logger.js]: Everything looks fine except this.
I Think the format method should only handle making it in a json format.

File: lib/logger.js [hunk 2]
```diff
@@ -21,16 +22,16 @@ Logger.prototype.setLogLevel = function (logLevel) {
   this.logLevel = logLevel
 }
 
-Logger.prototype.log = function (level, correlationId, message) {
+Logger.prototype.log = function (level, message) {
```
Comment (taltaub22) [inline lib/logger.js]: Do we want people to not be able to set the correlationId by them selves?

Comment (idanstark42) [reply lib/logger.js]: If they do we need to open this option in the `ms-correlation` package, The correlationId is a constant, and should only be created once per request. Putting this in a single location is a good practice in these cases

File: lib/logger.js (hunk 1 above)
Comment (idanstark42) [reply lib/logger.js]: Very well. I will change that

File: lib/logger.js (hunk 2 above)
Comment (taltaub22) [reply lib/logger.js]: Ok, great

**Pull request merged**